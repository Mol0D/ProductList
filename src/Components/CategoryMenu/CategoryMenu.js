import React from 'react';
import {ListGroup} from 'react-bootstrap';
import {Link} from 'react-router-dom';
import PropTypes from 'prop-types';

class CategoryMenu extends React.Component{
    render(){
        return(
            <nav>
                <ListGroup>
                    <ListGroup.Item>
                            <Link to="/products">
                                All
                            </Link>      
                    </ListGroup.Item>
                    {this.props.listMenu && this.props.listMenu.map((item, i) => (
                        <ListGroup.Item key={i}>
                            <Link to={`/products/${item}`}>
                                {item}
                            </Link> 
                        </ListGroup.Item>
                    ))} 
                </ListGroup>
            </nav> 
        )
    }
        
    }

CategoryMenu.propTypes ={
    listMenu: PropTypes.array
}


export default CategoryMenu;