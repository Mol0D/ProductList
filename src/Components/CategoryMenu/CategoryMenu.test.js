import React from 'react';
import {shallow, mount} from 'enzyme';
import CategoryMenu from './CategoryMenu';
import {BrowserRouter} from 'react-router-dom';


describe("CategoryMenu", ()=>{
    
    const props = {
        listMenu: ['aaa', 'bbb', 'ccc']
    }

    it("without component", ()=>{
        const wrapper = shallow(<CategoryMenu/>);
        expect(wrapper).toMatchSnapshot();
    })

    it("with component", ()=>{
        const wrapper = shallow(<CategoryMenu {...props}/>);
        expect(wrapper).toMatchSnapshot();
    })

    it("correct Prop", ()=>{
        const wrapper = mount(
            <BrowserRouter listMenu={props.listMenu}>
                <CategoryMenu/>)
            </BrowserRouter>
        );
        expect(wrapper.prop('listMenu')).toEqual(['aaa', 'bbb', 'ccc'])
    })

    
})