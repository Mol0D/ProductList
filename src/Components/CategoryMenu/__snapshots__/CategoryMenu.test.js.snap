// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryMenu with component 1`] = `ShallowWrapper {}`;

exports[`CategoryMenu without component 1`] = `ShallowWrapper {}`;
